src/main/java/picard/sam/SamAlignmentMerger.java

```diff
@@ -158,6 +141,70 @@ public void mergeAlignment(final File referenceFasta) {
         }
     }
 
+    @Override
+    protected SAMSequenceDictionary getDictionaryForMergedBam() {
+        SAMSequenceDictionary finalDict = null;
+        if (alignedSamFile != null && !alignedSamFile.isEmpty()) {
+            for (final File alignedFile : alignedSamFile) {
+                final SAMSequenceDictionary dict = SAMSequenceDictionaryExtractor.extractDictionary(alignedFile);
+                if (finalDict == null) {
+                    finalDict = dict;
+                } else {
+                    if (!compareSequenceDictionaries(finalDict, dict)) {
+                        log.error("Dictionaries of input files differ. Unable to continue.");
+                        throw new UnsupportedOperationException("Cannot use input files with different dictionaries.");
+                    }
+                }
+            }
+        } else {
+            final SeparateEndAlignmentIterator mergingIterator = new SeparateEndAlignmentIterator(this.read1AlignedSamFile, this.read2AlignedSamFile, referenceFasta);
+            finalDict = mergingIterator.getHeader().getSequenceDictionary();
+        }
+
+        SAMSequenceDictionary referenceDict = SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta);
```

Comment:
I would be tempted to do this _first_ and then compare the other dictionaries from the SAM files to this one.